Tests/NSURLUtilsTest.m

```diff
@@ -63,59 +46,18 @@ - (void)testQueryStringToDictionary {
 	XCTAssertEqualObjects(@"value3=more", [dict2 objectForKey:@"key3"]);
 }
 
-- (void)testDeriveWithQuery {
-	NSURL *URL = [NSURL URLWithString:@"http://api.yelp.com/path?key1=value1&key2=value2"];
-	NSURL *derivedURL = [URL gh_deriveWithQuery:@"key3=value3&key4=value4"];
-	XCTAssertEqualObjects([derivedURL description], @"http://api.yelp.com/path?key3=value3&key4=value4");	
-}
-
-- (void)testDeriveComplexWithQuery {
-	NSURL *URL = [NSURL URLWithString:@"https://user:pass@api.yelp.com:400/path?key1=value1&key2=value2#myfrag"];
-	NSURL *derivedURL = [URL gh_deriveWithQuery:@"key3=value3&key4=value4"];
-	XCTAssertEqualObjects([derivedURL description], @"https://user:pass@api.yelp.com:400/path?key3=value3&key4=value4#myfrag");	
-}
-
-- (void)testDeriveWithQueryUnicodeURL {
-  NSString *URLString = @"http://api.yelp.com/events/%E6%B8%AF%E5%8C%BA-%E3%81%8B%E3%81%8D%E6%B0%B7%E3%82%B3%E3%83%AC%E3%82%AF%E3%82%B7%E3%83%A7%E3%83%B3-copen-local-base-roppongi";
-  NSURL *URL = [NSURL URLWithString:URLString];
-  NSURL *derivedURL = [URL gh_deriveWithQuery:nil];
-  XCTAssertEqualObjects(URLString, [derivedURL description]);
-}
-
-- (void)testDeriveWithQueryRedirectToEscapedURL {
-  NSURL *URL = [NSURL URLWithString:@"http://www.yelp.com/redir"];
-  NSURL *derivedURL = [URL gh_deriveWithQuery:@"url=http%3A%2F%2Fwww.google.com"];
-  XCTAssertEqualObjects([derivedURL description], @"http://www.yelp.com/redir?url=http%3A%2F%2Fwww.google.com");
-}
-
-- (void)testCanonical {
-	NSURL *URL = [NSURL URLWithString:@"https://user:pass@api.yelp.com:400/path?b=c&a=d#myfrag"];
-	NSURL *canonical = [URL gh_canonical];
-	XCTAssertEqualObjects(canonical, [NSURL URLWithString:@"https://user:pass@api.yelp.com:400/path?a=d&b=c#myfrag"]);
-
-	NSURL *URL2 = [NSURL URLWithString:@"https://user:pass@api.yelp.com:400/path?b=c&a=d&ignore=ignored#myfrag"];
-	NSURL *canonical2 = [URL2 gh_canonicalWithIgnore:[NSArray arrayWithObject:@"ignore"]];
-	XCTAssertEqualObjects(canonical2, [NSURL URLWithString:@"https://user:pass@api.yelp.com:400/path?a=d&b=c#myfrag"]);
-}
-
-- (void)testFilter {
-  NSURL *URL = [NSURL URLWithString:@"https://user:pass@api.yelp.com:400/path?b=c&a=d&ignore=ignored#myfrag"];
-	NSURL *filtered = [URL gh_filterQueryParams:[NSArray arrayWithObjects:@"ignore", @"ignore2", nil] sort:YES];
-	XCTAssertEqualObjects(filtered, [NSURL URLWithString:@"https://user:pass@api.yelp.com:400/path?a=d&b=c#myfrag"]);
-}
-
 - (void)testQueryDictionaryWithArray {
 	NSArray *array1 = [NSArray arrayWithObjects:@"va", @"vb", @"vc", nil];
 	NSDictionary *dict = [NSDictionary dictionaryWithObjectsAndKeys:array1, @"key1", @"value2", @"key2", nil];
-	NSString *s = [NSURL gh_dictionaryToQueryString:dict sort:YES];
-	XCTAssertEqualObjects(s, @"key1=va%2Cvb%2Cvc&key2=value2");	
+	NSString *s = [NSURL gh_dictionaryToQueryString:dict];
+	XCTAssertEqualObjects(s, @"key1=va,vb,vc&key2=value2");	
```

Comment:
this behavior is the only thing i'm concerned about. while correct, this is technically a breaking change – it shouldn't impact anything like signatures (those were always calculated from the unescaped version anyway) but will, for example, mean potential cache misses: any URL with a collection query parameter like this would have been cached with a query string with escaped commas, and this will now fail to find that cached result.

the bigger concern imo is "does any API that consumes these collections _require_ them to be escaped? will anything break because of this?" my suspicion is no but i'm not 100% sure…